uc-crux-llvm/src/UCCrux/LLVM/Classify.hs

```diff
@@ -445,14 +455,33 @@ classifyBadBehavior appCtx modCtx funCtx sym skipped simError (Crucible.RegMap _
                       -- variable.
                       if Set.null skipped
                         then truePositive (ReadUninitializedStack loc)
-                        else unclass appCtx badBehavior
+                        else unclass appCtx simError badBehavior
                   Just (G.AllocInfo G.HeapAlloc _sz _mut _align loc) ->
                     if loc == mallocLocation
-                      then unclass appCtx badBehavior
+                      then unclass appCtx simError badBehavior
                       else truePositive (ReadUninitializedHeap loc)
-                  -- TODO(lb): Result in false negatives
-                  -- Just (G.AllocInfo G.GlobalAlloc _sz _mut _align _loc) ->
-                  --   unfixed appCtx UnfixedUninitializedGlobal
+                  Just (G.AllocInfo G.GlobalAlloc _sz _mut _align _loc) ->
+                    case flip Map.lookup (LLVMMem.memImplSymbolMap memImpl) =<< blk of
+                      Nothing -> requirePossiblePointer ReadNonPointer ptr
+                      Just glob ->
+                        case ( makeDeclSymbol glob (modCtx ^. declTypes),
+                               makeGlobalSymbol (modCtx ^. globalTypes) glob
+                             ) of
+                          (Just {}, _) -> truePositive (DerefFunctionPointer glob)
+                          (_, Just gSymb) ->
+                            case modCtx ^. globalTypes . globalSymbol gSymb of
+                              Some FTUnboundedArrayRepr {} ->
+                                unimplemented
+                                  "classify"
+                                  Unimplemented.NonEmptyUnboundedSizeArrays
+                              _ ->
+                                panic
```

Comment:
Unfortunately, I'm not hugely confident that this can never happen, but I'm also not exactly sure what it would mean if it did, or how to trigger. Before merging, I'll run the tool over some large codebases to be sure it doesn't come up in practice.

Reply:
Alright, it's definitely possible:
```
[Crux] You have encountered a bug in crux-llvm bugfinding mode's implementation.
*** Please create an issue at https://github.com/GaloisInc/crucible/issues

%< --------------------------------------------------- 
  Revision:  9b8c591e76d2387b9a3c040cb4c97269a81d3591
  Branch:    lb/populate-globals (uncommited files present)
  Location:  classify
  Message:   Weird error - read from bad offset of global?
CallStack (from HasCallStack):
  panic, called at src/UCCrux/LLVM/Errors/Panic.hs:37:9 in uc-crux-llvm-0.1-inplace:UCCrux.LLVM.Errors.Panic
  panic, called at src/UCCrux/LLVM/Classify.hs:478:33 in uc-crux-llvm-0.1-inplace:UCCrux.LLVM.Classify
%< --------------------------------------------------- 
```
Let's find out what's going on!

Reply:
It occurs in the function `buildargv` in `addr2line`:
```
[Crux] Exploring buildargv
[Crux] Checking function buildargv
[Crux] Running buildargv on arguments...
[Crux] Argument #0 input: symbolic integer:  width = 64
[Crux] Attempting to prove verification conditions.
[Crux] Explaining error: No previous write to this location was found
  Attempting load at type: i8
  Via pointer: annotation Nonce {indexValue = 1101} cfresh90@1098:bv
[Crux] Diagnosis:  Read from an uninitialized pointerin argument #0 at <top>
[Crux] Prescription: Add a precondition that the pointer points to initialized memory.
[Crux] Explaining error: No previous write to this location was found
  Attempting load at type: i8
  Via pointer: annotation Nonce {indexValue = 1101} cfresh90@1098:bv
[Crux] Diagnosis:  Read from an uninitialized pointerin argument #0 at <top>
[Crux] Prescription: Add a precondition that the pointer points to initialized memory.
[Crux] Total paths explored: 2
[Crux] New preconditions:
[Crux] Arguments:
  A pointer to initialized space for 1 elements: 
Globals:
Return values of skipped functions:
[Crux] Running buildargv on arguments...
[Crux] Argument #0 input: pointer with symbolic offset
[Crux] Attempting to prove verification conditions.
[Crux] Explaining error: No previous write to this location was found
  Attempting load at type: i16
  Via pointer: (233, let -- ./argv.c:121:10
                         v1141 = bvZext 64 (annotation Nonce {indexValue = 1107} cfresh90@1104:bv)
   in bvSum (bvMul 0x2:[64] v1141))
[Crux] You have encountered a bug in crux-llvm bugfinding mode's implementation.
```

Reply:
Here's the definition of that function at this revision (2.35.1): https://github.com/bminor/binutils-gdb/blob/7e46a74aa3713c563940960e361e08defda019c2/libiberty/argv.c#L176

Reply:
I've changed this to be "unclassified", pending further investigation.